### PR TITLE
general improvements

### PR DESCRIPTION
**Pull request by tesinormed**
yes

### CODE REVIEW

**Comment by tesinormed** [comment]
have not tested that my changes actually work ~~because i dont trust the program~~ so if the batch file or the program stops working please tell me

**Review by jojo2357 — APPROVED** [review]
alright lo

File: src/com/github/jojo2357/GitHubDataParser.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Scanner;
+
```
Comment (jojo2357) [inline src/com/github/jojo2357/GitHubDataParser.java]: import com.github.jojo2357.TimeStamp;//?

File: viewsgetter [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# repo finder extracts the repo names so that we can do something with it
+javac -d out src/com/github/jojo2357/*.java
+java -cp $PWD/out/ com.github.jojo2357.RepoRefiner
+
+# for each repo, we get traffic, and then parse that data
+while read p; do
+  curl "https://api.github.com/repos/%user%/%%x/traffic/views" -u $USER:$PASSWORD >$PWD\$p.txt
+  java -cp $PWD/out/ com.github.jojo2357.GitHubDataParser $p Views
+  curl "https://api.github.com/repos/%user%/%%x/traffic/clones" -u $USER:$PASSWORD >$PWD\$p.txt
+  java -cp $PWD/out/ com.github.jojo2357.GitHubDataParser $p Clones
```
Comment (jojo2357) [inline viewsgetter:26]: remove space after out/ ?

File: viewsgetter.bat [hunk 3]
```diff
@@ -1,20 +1,20 @@
-REM username
+@rem username
```
Comment (jojo2357) [inline viewsgetter.bat]: Um i dont think that this is how it works for windows...REM has been working so idk

File: src/com/github/jojo2357/RepoRefiner.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.github.jojo2357;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class RepoRefiner {
+    public static void main(String[] args) throws IOException {
```
Comment (jojo2357) [inline src/com/github/jojo2357/RepoRefiner.java]: remove throws (i can do this, just making a note)

File: src/com/github/jojo2357/GitHubDataParser.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        }
+        inputReader.close();
+        System.out.println(inputFile.delete());
+    }
+
+    /**
+     * refines the views/unique views lines since they all are the same format
+     *
+     * @param in the views/unique views lines
+     * @return the refined views/unique views lines
```
Comment (jojo2357) [inline src/com/github/jojo2357/GitHubDataParser.java]: do this everywhere?

File: viewsgetter.bat [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
 set user=""
```
Comment (jojo2357) [inline viewsgetter.bat]: change to be parameters (arguments)

File: README.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# Github Views Logger  
+This is a mix of java and batch scripting that gets pertinent data from a user's repos and stores them.   
+It is set up to get and store a day-by-day count of views and clones that does not simply go away after 24 hours.  
+I'm not a superhero that can retrieve data that is more than 2 weeks old unless it has already been stored by the program, then it merely keeps going from the first time logging started.  
+
+## REQUIRED STEPS:  
+go into ViewsGetter.bat and in the quotes, add a username, and the password (see note at end)
```
Comment (jojo2357) [inline README.md:7]: change this when we change the windows file

**Review by jojo2357 — CHANGES_REQUESTED** [review]
Thanks bro, some things that need to be changed, but it looks good

File: src/com/github/jojo2357/GitHubDataParser.java (hunk 1 above)
Comment (tesinormed) [reply src/com/github/jojo2357/GitHubDataParser.java]: is it missing?

File: viewsgetter (hunk 2 above)
Comment (tesinormed) [reply viewsgetter:26]: thats setting the classpath

File: src/com/github/jojo2357/RepoRefiner.java (hunk 4 above)
Comment (tesinormed) [reply src/com/github/jojo2357/RepoRefiner.java]: aight i guess ill remove throws

File: viewsgetter.bat (hunk 6 above)
Comment (tesinormed) [reply viewsgetter.bat]: ok ill try and figure it out

File: README.md (hunk 7 above)
Comment (tesinormed) [reply README.md:7]: should i make 2 sections for linux and windows?

File: viewsgetter (hunk 2 above)
Comment (jojo2357) [reply viewsgetter:26]: very well it just seemed off

File: src/com/github/jojo2357/RepoRefiner.java (hunk 4 above)
Comment (jojo2357) [reply src/com/github/jojo2357/RepoRefiner.java]: i can do it dw

File: viewsgetter.bat (hunk 6 above)
Comment (jojo2357) [reply viewsgetter.bat]: i can do this dw

File: viewsgetter.bat (hunk 3 above)
Comment (tesinormed) [reply viewsgetter.bat]: aight ig

File: viewsgetter.bat (hunk 6 above)
Comment (tesinormed) [reply viewsgetter.bat]: aight

File: src/com/github/jojo2357/GitHubDataParser.java (hunk 1 above)
Comment (jojo2357) [reply src/com/github/jojo2357/GitHubDataParser.java]: i think so. i dont see it and i think that it should be there.

File: viewsgetter.bat (hunk 3 above)
Comment (tesinormed) [reply viewsgetter.bat]: done

File: src/com/github/jojo2357/GitHubDataParser.java (hunk 5 above)
Comment (tesinormed) [reply src/com/github/jojo2357/GitHubDataParser.java]: sure

File: README.md (hunk 7 above)
Comment (jojo2357) [reply README.md:7]: yeah but we can do that once we finish this merge thing

File: src/com/github/jojo2357/GitHubDataParser.java (hunk 1 above)
Comment (tesinormed) [reply src/com/github/jojo2357/GitHubDataParser.java]: you dont need to import if its in the same package

**Review by jojo2357 — APPROVED** [review]
all good! lets get cracking on the next round!

**Review by jojo2357 — APPROVED** [review]
All good!

**Review by jojo2357 — APPROVED** [review]
good enough now i gotta figure out how this github works

**Pull request merged**